arcaflow_plugin_rtla/rtla_schema.py

```diff
@@ -0,0 +1,145 @@
+        else:
+            result.append(f"-{key}")
+            result.append(str(value))
+    return result
+
+
+@dataclass
+class TimerlatInputParams:
+    period: typing.Annotated[
+        typing.Optional[int],
+        schema.name("timerlat period"),
+        schema.description("Timerlat period in μs"),
+    ] = None
+    cpus: typing.Annotated[
+        typing.Optional[typing.List[int]],
+        schema.name("cpus"),
+        schema.description("Run the tracer only on the given cpus"),
+    ] = None
+    house_keeping: typing.Annotated[
+        typing.Optional[typing.List[int]],
+        schema.id("house-keeping"),
+        schema.name("house-keeping cpus"),
+        schema.description("Run rtla control threads only on the given cpus"),
+    ] = None
+    duration: typing.Annotated[
+        typing.Optional[int],
+        schema.name("timerlat duration seconds"),
+        schema.description("Duration of the session in seconds"),
+    ] = None
+    nano: typing.Annotated[
+        typing.Optional[bool],
+        schema.name("display nanoseconds"),
+        schema.description("Display data in nanoseconds"),
+    ] = None
+    bucket_size: typing.Annotated[
+        typing.Optional[int],
+        schema.id("bucket-size"),
+        schema.name("histogram bucket size"),
+        schema.description("Set the histogram bucket size (default 1)"),
+    ] = None
+    entries: typing.Annotated[
+        typing.Optional[int],
+        schema.name("histogram entries"),
+        schema.description("Set the number of entries of the histogram (default 256)"),
+    ] = None
+    user_threads: typing.Annotated[
+        typing.Optional[bool],
+        schema.id("user-threads"),
+        schema.name("use user threads"),
+        schema.description(
+            "Use rtla user-space threads instead of kernel-space timerlat threads"
+        ),
+    ] = None
+
+    def to_flags(self) -> str:
+        return params_to_flags(
+            {
+                "p": self.period,
+                "c": self.cpus,
+                "H": self.house_keeping,
+                "d": self.duration,
+                "n": self.nano,
+                "b": self.bucket_size,
+                "E": self.entries,
+                "u": self.user_threads,
+            }
+        )
+
+
+@dataclass
+class LatencyStats:
+    count: typing.Annotated[
+        int,
+        schema.name("number of measurements"),
+        schema.description("Number of latency measurements"),
+    ] = None
+    min: typing.Annotated[
+        int,
+        schema.name("minimum latency"),
+        schema.description("Minimum latency value"),
+    ] = None
+    avg: typing.Annotated[
+        int,
+        schema.name("average latency"),
+        schema.description("Average latency value"),
+    ] = None
+    max: typing.Annotated[
+        int,
+        schema.name("maximum latency"),
+        schema.description("Maximum latency value"),
+    ] = None
```

Comment:
Should this mention the units?

Reply:
This made me realize that the inputs allow us to change the unit, so I added that as a separate object in the output. Since this can be either ms or us based on the input, I can't state the unit directly in the description except maybe as an either/or.